### ISSUE TITLE
No need for the embedding layer to be included in backprop 

### ISSUE THREAD

**Issue by vishalgolcha** [bentrevett/pytorch-sentiment-analysis#16]
Since you are loading already trained vectors into the model we can set require_grad parameter as false for the embedding layer , right ? There is no need to propagate the gradients for them . 
This could create an issue especially if you encounter a word you haven't seen in the training data in your  vaildation .

**Comment by bentrevett** [comment]
Yes, you can leave the word embeddings as they are, however it is common to "fine-tune" them on your task, which is what we do in the tutorials.

I'll add a line or two about this to the tutorials.

**Comment by vishalgolcha** [comment]
One more thing shouldn't the vocaulary builder should include train , test and vaildation words  that is the whole vocabulary and not just training .

**Comment by bentrevett** [comment]
No, your vocabulary should only be build from the training set. You want your validation/test sets to reflect the real world usage of this system as much as possible. 

In the real world, you'll be getting words that are not in your vocabulary, so by having this also true of your validation/test sets, this means the val/test metrics are more reflective of the actual real world performance.
